plutus-core/plutus-ir/test/transform/retainedSize/builtinBinding.golden

```diff
@@ -0,0 +1,9 @@
+(let
+  (nonrec)
+  (termbind
+    (strict)
+    (vardecl $0$ add (fun (con integer) (con integer)))
+    (builtin addInteger)
+  )
+  (abs a (type) (lam x a x))
+)
```

Comment:
`add` is not used, hence it retains zero size.

Reply:
Hang on, this seems wrong. `add` itself is dead, so it's not retained *by* the root, but the `add` binding itself retains its RHS. That is, looking at this program, you could make it smaller by deleting `add`.

Reply:
I can't obviously see where we make it so that dead bindings retain nothing, but I don't think we should.

Reply:
> Hang on, this seems wrong. add itself is dead, so it's not retained by the root, but the add binding itself retains its RHS. That is, looking at this program, you could make it smaller by deleting add.

That zero means that `add` is not needed at all and should have been eliminated at an earlier stage. If it was the size of the RHS, `add` would look identical to bindings that are actually retained and so we'd lose some quite important visual info.

But it might be worth displaying this info differently though. Like "not needed and retains `N`". Do you have suggestions on how to display this info best?

> I can't obviously see where we make it so that dead bindings retain nothing, but I don't think we should.

[Here](https://github.com/input-output-hk/plutus/blob/c4f2f1638d11ee5b23eb3474fbbfc1a6113621f0/plutus-core/plutus-ir/src/PlutusIR/Analysis/RetainedSize.hs#L214-L215).

Reply:
IMO this is confusing two things: how much does this binding retain, and is it dead?

> If it was the size of the RHS, add would look identical to bindings that are actually retained and so we'd lose some quite important visual info.

Would we? As it is, we don't know how much `add` retains, instead we get the *different* information that it's dead!

> Here.

That just pushes my question back to "why don't dead things get in the map?"